src/utility.jl

```diff
@@ -462,101 +462,46 @@ function round_sol(m::Optimizer, relaxed_sol)
 end
 
 function fetch_mip_solver_identifier(m::Optimizer;override="")
-
-    isempty(override) ? solverstring = string(Alp.get_option(m, :mip_solver)) : solverstring = override
-
-   # Higher-level solvers: that can use sub-solvers
-   if occursin("Pajarito", solverstring)
-       m.mip_solver_id = "Pajarito"
-      return
-   elseif occursin("Pavito", solverstring)
-       m.mip_solver_id = "Pavito"
-      return
-   elseif occursin("Juniper", solverstring)
-       m.mip_solver_id = "Juniper"
-      return
-   end
-
-   # Lower level solvers
-   if occursin("Gurobi", solverstring)
-       m.mip_solver_id = "Gurobi"
-   elseif occursin("CPLEX", solverstring)
-       m.mip_solver_id = "Cplex"
-   elseif occursin("Cbc", solverstring) # /!\ the `SolverName()` is "COIN Branch-and-Cut (Cbc)"
-       m.mip_solver_id = "Cbc"
-   elseif occursin("GLPK", solverstring)
-       m.mip_solver_id = "GLPK"
-   elseif occursin("HiGHS", solverstring)
-       m.mip_solver_id = "HiGHS"
-   elseif occursin("Xpress", solverstring)
-        m.mip_solver_id = "Xpress" 
-   else
-      error("Unsupported MIP solver $solverstring; use a Alpine-supported MIP solver")
-   end
-
-   return
+    mip_solver = Alp.get_option(m, :mip_solver).optimizer_constructor
+    if typeof(mip_solver) !== DataType
+        mip_solver = typeof(mip_solver)
+    end
```

Comment:
@odow @blegat Gurobi specific attributes are necessary and hence had to leave this part. But I am sure this wasn't the best way to handle when Gurobi is specified with `GRB_ENV` environment. Just checking for `.optimizer_constructor` seems to give some random number. So going through this hoop. Let me know if there is a better way.

Reply:
You should wait for the MIP solver to be instantiated and then get SolverName and use that so set the id